README.md

```diff
@@ -1 +1,212 @@
+(_%package/path/to/file.json_, for example).  You can set this using an environment variable, a command line
+parameter, or by specifying it in the defaults, as in:
+
+```
+fluid.defaults("my.launcher", {
+    gradeNames: ["gpii.launcher"],
+    yargsOptions: {
+        defaults: {
+            optionsFile: "%my-package/configs/config.json"
+        }
+    }
+});
+```
+
+The contents of this file will be loaded using the [configuration loading built into kettle](https://github.com/fluid-project/kettle/blob/master/docs/ConfigsAndApplications.md).  The file
+is expected to correspond roughly to a component definition, but supports additional options for including other configuration files.  See [the kettle documentation](https://github.com/fluid-project/kettle/blob/master/docs/ConfigsAndApplications.md#structure-of-a-kettle-config
+) for more details.
+
+
+# Example Usage
+
+To use the launcher, you need to define and launch a `gpii.launcher` instance, as in the example included with this package.
+
+
+```
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+fluid.setLogging(true);
+
+var my = fluid.registerNamespace("my");
+fluid.require("%gpii-launcher");
+
+fluid.defaults("my.launcher.worker", {
+    gradeNames: ["fluid.component"],
+    var1:    "set in the component",
+    listeners: {
+        "onCreate.log": {
+            funcName: "fluid.log",
+            args: ["Var 1:", "{that}.options.var1"]
+        },
+        "onCreate.destroy": {
+            func: "{that}.destroy",
+            priority: "after:log"
+        }
+    }
+});
+
+fluid.registerNamespace("my.launcher");
+
+fluid.defaults("my.launcher", {
+    gradeNames: ["gpii.launcher"],
+    yargsOptions: {
+        describe: {
+            "var1": "you can set this option"
+        },
+        defaults: {
+            "optionsFile": "%gpii-launcher/examples/my-launcher-config.json"
+        }
+    }
+});
+
+my.launcher();
+```
+
+In this example, we set a default for `optionsFile`, which is used to load a sample configuration file:
+
+```
+{
+  "type": "launcherConfig",
+  "options": {
+    "gradeNames": ["my.launcher.worker"],
+    "var1": "set in the options file."
+  }
+}
+```
+
+Here are some examples of the output that results from using the above with various combinations of command line
+parameters and environment variables on a UNIX-like system:
+
+```
+$ node examples/my-launcher.js
+Var 1: Set in the options file.
+
+$ node examples/my-launcher.js --var1 "Set from the command line."
+Var 1: Set from the command line.
+
+$ var1="Set by an environment variable." node examples/my-launcher.js
+Var 1: Set by an environment variable
+
+$ node examples/my-launcher.js --optionsFile "%gpii-launcher/examples/my-alternate-launcher-config.json"
+Var 1: Set in the alternate options file.
+
+```
+# Referencing Deep Variables
+
+By default, yargs supports [using "dot notation" to refer to deep variables](http://yargs.js.org/docs/#parsing-tricks-dot-notation).
```

Comment:
It's not clear how, if at all, this interacts with Infusion's own idiom for "deep objects". It would be nice for there to be support in the "universal driver" for pushing arbitrary options into arbitrary grades in the system purely via the command line.